modules/evaluation/src/main/PlayerAggregateAssessment.scala

```diff
@@ -97,19 +98,38 @@ case class PlayerAggregateAssessment(
     val avg = listAverage(playerAssessments.filter(predicate).map(_.sfAvg)).toInt
     if (playerAssessments.exists(predicate)) Some(avg) else none
   }
+  
+  def sfCiGiven(predicate: PlayerAssessment => Boolean): Option[Array[Int]] = {
```

Comment:
There, filtered list `playerAssessments.filter(predicate)` is computed 3 times (4 including `playerAssessments.exists(predicate)`) per each call `sfCiGiven`.

Filtered list of avg `playerAssessments.filter(predicate).map(_.sfAvg)` is computed twice per each call.

`Array` being a mutable type, I would say to should only be used in API (parameters, return type, of none private functions) when an immutable type can hardly be used.
There, a fixed number of `Int` (2) is returned, so `Tuple2` is more more safe and appropriate.

I think case `if (n <= 1)` should not be interesting for `width` (over `== 1`), which is only used in case `if (playerAssessments.exists(predicate))` (so in case filtered list is not empty and its size > 0).

```scala
def sfCiGiven(predicate: PlayerAssessment => Boolean): Option[(Int, Int)] = {
  val filteredAssessments = playerAssessments.filter(predicate)

  if (filteredAssessments.isEmpty) {
    none
  } else {
    val filteredSfAvg = filteredAssessments.map(_.sfAvg)
    val avg = listAverage(filteredSfAvg)
    val n = filteredAssessments.size

    val width = if (n == 1) 196 else listDeviation(filteredSfAvg) / sqrt(n) * 1.96

    Some((avg - width).toInt -> (avg + width).toInt)
  }
}
```

Reply:
Thank you for your comments. I will make those changes.
Additionally, I am now thinking that probably instead of creating this new function  sfCiGiven, I should actually edit sfAvgGiven to return an Option[(Int, Int, Int)], since for all 6 cases where the former is called, we also call the latter.
Also, Bessel's correction should actually be applied (1/sqrt(n - 1) instead of 1/sqrt(n)), so if (n == 1) empty. That way, the assessments would not even be shown for only 1 count, which is probably a good idea anyway.